src/main/java/net/snowflake/client/core/SFBaseSession.java

```diff
@@ -788,4 +791,18 @@ public String getQueryContext() {
   public void setQueryContext(String queryContext) {
     this.queryContext = queryContext;
   }
+
+  /**
+   * If true, JDBC will enable returning TIMESTAMP_WITH_TIMEZONE as column type, otherwise it will
+   * not. This function will always return true for JDBC client, so that the client JDBC will not
+   * have any behavior change. Stored proc JDBC will override this function to return the value of
+   * SP_JDBC_ENABLE_TIMESTAMP_WITH_TIMEZONE from server for backward compatibility.
+   */
+  public boolean getEnableReturnTimestampWithTimeZone() {
+    return enableReturnTimestampWithTimeZone;
+  }
+
+  public void setEnableReturnTimestampWithTimeZoneForTesting(boolean value) {
```

Comment:
This is public for testing, I think it's ok to leave it public because the api name included `forTesting`. The test which used this api is not in the same package with SFBaseSession so setting to protected makes it difficult to call this api in test.

Reply:
We also have similar practice in `setMemoryLimitForTesting`

Reply:
If we are adding test methods in actual driver jar, then we need to remove those. You need to test the way user is going to enable this feature. Please don't add methods just for testing in real production code. I discuss this with Sashi and we shouldn't add testing method.

Reply:
@sfc-gh-zli Can you override this function using a subclass in your test?